### PR TITLE
PHP 5.6: Use argument unpacking

### PR DESCRIPTION
**Pull request by Majkl578**
Depends on  #79 (composer.json & Travis).

### CODE REVIEW

File: src/DI/ContainerLoader.php [hunk 1]
```diff
@@ -108,7 +108,7 @@ protected function generate($class, $generator)
 	{
 		$compiler = new Compiler;
 		$compiler->getContainerBuilder()->setClassName($class);
-		$code = call_user_func_array($generator, [& $compiler]);
+		$code = $generator(...[& $compiler]);
```
Comment (hrach) [inline src/DI/ContainerLoader.php]: `$generator(& $compiler)` ?

Comment (JanTvrdik) [inline src/DI/ContainerLoader.php]: removing `call_user_func_*` will not work in all cases, e.g. if `$generator` is string.

Comment (Majkl578) [inline src/DI/ContainerLoader.php]: @hrach: That is a syntax error in PHP 7.
What format exactly? I think only static methods? That would be eventually solved when we switch to PHP 7.

Comment (hrach) [inline src/DI/ContainerLoader.php]: Oh I see, this is a nasty hack to workaround call-time pass-by-reference. Imo we should get rid of it anyway :)

Comment (dg) [inline src/DI/ContainerLoader.php]: Nice hack, but for PHP 5.6 is `call_user_func_array` required here.

Comment (Majkl578) [inline src/DI/ContainerLoader.php]: Ok, reverted & rebased.

**Pull request merged**